File: DesignPatternsExercises/java/MediatorDesignPattern/Exercise/Buyer.java
public class Buyer {
	private Mediator mediator;
	private String currency;

	public Buyer(Mediator mediator, String currency) {
		this.mediator = mediator;
		this.currency = currency;
		mediator.__________(1)__________;
	}

	public void __________(2)__________(float bid) {
		System.out.println("Buyer is attempting a bid of " + currency + " " + bid);
		mediator.placeBid(currency, bid);
	}
}

File: DesignPatternsExercises/ruby/mediator_design_pattern/exercise/buyer.rb
class Buyer
  def initialize(mediator, currency)
    @mediator = mediator
    @currency = currency
    mediator.__________(1)__________
  end

  def __________(2)__________(bid)
    puts "Buyer is attempting a bid of #{@currency} #{bid}"
    @mediator.place_bid(@currency, bid)
  end
end
